pyquil/wavefunction.py

```diff
@@ -33,8 +33,8 @@ def __init__(self, amplitude_vector, classical_memory=None):
         if len(amplitude_vector) == 0 or len(amplitude_vector) & (len(amplitude_vector) - 1) != 0:
             raise TypeError("Amplitude vector must have a length that is a power of two")
 
-        self.amplitudes = amplitude_vector
-        self.classical_memory = classical_memory
+        self.amplitudes = np.asarray(amplitude_vector)
+        self.classical_memory = np.asarray(classical_memory)
```

Comment:
while I also prefer this to be the type, this does signify a schema change (and incidentally causes tests to fail). @stevenheidel what is your policy?

Reply:
I also agree with this change, but we can't do it in a point release since it will break existing code. Add it to the https://github.com/rigetticomputing/pyquil/issues/227

Reply:
I figured it was already mainly numpy arrays because `from_packed_string` passes a numpy array to the constructor

Reply:
If it's already numpy arrays then the above discussion doesn't apply

Reply:
`Wavefunction.zeros()` returned a wf with a list, but then that was already inconsistent and this will improve the behavior.